src/main/java/frc/trigon/robot/poseestimation/robotposesources/RobotPoseSource.java

```diff
@@ -13,75 +16,96 @@ public class RobotPoseSource {
 
         if (Robot.IS_REAL)
-            robotPoseSourceIO = robotPoseSourceType.createIOFunction.apply(name, robotCenterToCamera);
+            robotPoseSourceIO = robotPoseSourceType.createIOFunction.apply(name);
         else
             robotPoseSourceIO = new RobotPoseSourceIO();
     }
 
-    public static double[] pose3dToDoubleArray(Pose3d pose) {
-        if (pose == null)
-            return new double[0];
-
-        return new double[]{
-                pose.getTranslation().getX(),
-                pose.getTranslation().getY(),
-                pose.getTranslation().getZ(),
-                pose.getRotation().getX(),
-                pose.getRotation().getY(),
-                pose.getRotation().getZ()
-        };
-    }
-
     public void update() {
         robotPoseSourceIO.updateInputs(inputs);
         Logger.processInputs("Cameras/" + name, inputs);
-        cachedPose = getUnCachedRobotPose();
-        if (!inputs.hasResult || inputs.averageDistanceFromTags == 0 || cachedPose == null)
+
+        robotPose = getBestRobotPose(inputs.averageDistanceFromAllTags);
+
+        if (!inputs.hasResult || inputs.averageDistanceFromAllTags == 0 || robotPose == null)
             Logger.recordOutput("Poses/Robot/" + name + "Pose", RobotPoseSourceConstants.EMPTY_POSE_LIST);
         else
-            Logger.recordOutput("Poses/Robot/" + name + "Pose", cachedPose);
+            Logger.recordOutput("Poses/Robot/" + name + "Pose", robotPose);
     }
 
-    public int getVisibleTags() {
-        return inputs.visibleTags;
+    public boolean hasNewResult() {
+        return (inputs.hasResult && inputs.averageDistanceFromAllTags != 0) && isNewTimestamp();
     }
 
-    public double getAverageDistanceFromTags() {
-        return inputs.averageDistanceFromTags;
+    /**
+     * @return the current estimated robot position
+     */
+    public Pose2d getCurrentRobotPose() {
+        return robotPose;
     }
 
-    public boolean hasNewResult() {
-        return (inputs.hasResult && inputs.averageDistanceFromTags != 0) && isNewTimestamp();
+    public String getName() {
+        return name;
     }
 
-    public Pose2d getRobotPose() {
-        return cachedPose;
+    public double getAverageDistanceFromTags() {
+        return inputs.averageDistanceFromAllTags;
     }
 
-    public String getName() {
-        return name;
+    public int getNumberOfVisibleTags() {
+        return inputs.visibleTagIDs.length;
     }
 
     public double getLastResultTimestamp() {
         return inputs.lastResultTimestamp;
     }
 
-    private Pose2d getUnCachedRobotPose() {
-        final Pose3d cameraPose = doubleArrayToPose3d(inputs.cameraPose);
-        if (cameraPose == null)
-            return null;
+    /**
+     * Gets the best possible robot Pose from the camera's data
+     *
+     * @param averageDistanceFromAllTags the average of the distance from all visible tags
+     * @return the result
+     */
```

Comment:
stop writing `the result`, period at the end of a sentence in the general explanation, and explain how the whole logic works and how it will determine the best pose.